### PR TITLE
feat: EthL2s

### LINKED ISSUE #2089: Update 0082-ETHD-ethereum-data-source.md to details the Data sourcing from L2 ethereum oracles and add ACs
The ethereum oracle spec needs to be updated to detail the design around data sourcing from L2 ethereum oracles and add ton add any applicable ACs

- [ ] update the spec with the design
- [ ] add ACs for all existing ACs but to work with L2s
- [ ] add ACs for the set up and config of L2 data source
- [ ] add ACs for data sourcing to be persevered / not impacted over a protocol upgrade
- [ ] add ACs to ensure that when having multiple L2s there is no way data can be sourced from the incorrect chain

### PR DESCRIPTION
**Pull request by davidsiska-vega**
close https://github.com/vegaprotocol/specs/issues/2089

### CODE REVIEW

File: protocol/0087-EL2D-ethereumL2-data-source.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+## Decription 
+
+In addition to listening to Ethereum events and reading from Ethereum contrants as described in [Ethereum data source](./0082-ETHD-ethereum-data-source.md) it will be possible for Vega nodes to listen to events from and read from other Ethereum L2s. 
+
+The overarching principle is that the L2s incorporate the ethereum EVMs and thus contracts and ABIs are assumed to be functionally the same as on Ethereum itself. 
+
+## Registration / removal
+
+A new network parameter, a JSDON list of network id / chain id / name one per L2 will be used for setting supported L2. This is set via [governance](./0028-GOVE-governance.md).
```
Comment (gordsport) [inline protocol/0087-EL2D-ethereumL2-data-source.md]:
```suggestion
A new network parameter, a JSON list of network id / chain id / name one per L2 will be used for setting supported L2. This is set via [governance](./0028-GOVE-governance.md). 
```

File: protocol/0087-EL2D-ethereumL2-data-source.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+## Registration / removal
+
+A new network parameter, a JSDON list of network id / chain id / name one per L2 will be used for setting supported L2. This is set via [governance](./0028-GOVE-governance.md). 
+Name `blockchains.ethereumL2Configs`.
+Example value: 
+```
+{"configs": [{"network_id": 10, "chain_id": 10, "confirmations": 3, "name":"optimism"}]}
+```
+Duplicate values of "network_id": 10, "chain_id" or  "name":"optimism" or not allowed (an update will be rejected at validation stage).
```
Comment (gordsport) [inline protocol/0087-EL2D-ethereumL2-data-source.md]:
```suggestion
Duplicate values of "network_id": 10, "chain_id" or  "name":"optimism" are not allowed (an update will be rejected at validation stage).
```

File: protocol/0087-EL2D-ethereumL2-data-source.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+Any updat emust always change the entire JSON (it's not possible to change individual entries).
```
Comment (gordsport) [inline protocol/0087-EL2D-ethereumL2-data-source.md]:
```suggestion
Any update must always change the entire JSON (it's not possible to change individual entries).
```

File: protocol/0087-EVMD-eth-rpc-and-evm-data-source.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+A new network parameter, a JSON list of network id / chain id / name one per Ethereum RPC and EVM chain will be used for setting supported L2. This is set via [governance](./0028-GOVE-governance.md).
+Name `blockchains.ethereumL2Configs`.
+Example value:
+
+```json
+{"configs": [{"network_id": 10, "chain_id": 10, "confirmations": 3, "name":"optimism"}]}
+```
+
+Duplicate values of "network_id": 10, "chain_id" or  "name":"optimism" are not allowed (an update will be rejected at validation stage).
```
Comment (gordsport) [inline protocol/0087-EVMD-eth-rpc-and-evm-data-source.md]:
```suggestion
Duplicate values of "network_id", "chain_id" or  "name" are not allowed (an update will be rejected at validation stage).
```

Should these values be in here? suggested removal.

File: protocol/0087-EVMD-eth-rpc-and-evm-data-source.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+source: EthereumEvent {
+            source_chain_id: 0x789
+            contract: 0xDEADBEEF
+            ABI: "...JSON..."
+            event: "MyFaveEvent"
+```
+
+Then market 1 only sees events of that type from `EthRpcEvmCompatible` 0x123 while market 2 only sees events of that type from `EthRpcEvmCompatible` 0x789 (<a name="0087-EVMD-014" href="#0087-EVMD-014">0087-EVMD-014</a>).
+
+- It should be possible to use only `EthRpcEvmCompatible` data sources in a market proposal, or create any combination with any of the other types of currently existing external or internal data sources (<a name="0087-EVMD-015" href="#0087-EVMD-015">0087-EVMD-015</a>)
```
Comment (wwestgarth) [inline protocol/0087-EVMD-eth-rpc-and-evm-data-source.md:99]: These are all very future's focussed. Could we make sure we have a happy path AC for perpetuals? And maybe also one for doing a `UpdateMarket` on a perpetual from a data-source on normal Ethereum to a data-source from an EVM-compatible chain?

Comment (davidsiska-vega) [reply protocol/0087-EVMD-eth-rpc-and-evm-data-source.md:99]: Both very good points @wwestgarth let me add them.

File: protocol/0087-EVMD-eth-rpc-and-evm-data-source.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+## Description
+
+In addition to listening to Ethereum events and reading from Ethereum contracts as described in [Ethereum data source](./0082-ETHD-ethereum-data-source.md) it will be possible for Vega nodes to listen to events from and read from other chains that implement Ethereum RPC and run EVM, in particular Ethereum L2.
+
+The overarching principle is that the chain provides ethereum RPC / EVMs and thus contracts and ABIs are assumed to be functionally the same as on Ethereum itself.
+
+## Registration / removal
+
+A new network parameter, a JSON list of network id / chain id / name one per Ethereum RPC and EVM chain will be used for setting supported L2. This is set via [governance](./0028-GOVE-governance.md).
+Name `blockchains.ethereumL2Configs`.
```
Comment (davidsiska-vega) [inline protocol/0087-EVMD-eth-rpc-and-evm-data-source.md]: Rename this!

File: protocol/0087-EVMD-eth-rpc-and-evm-data-source.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+
+> [!TIP]
+> A Layer 2 blockchain refers to network protocols that are layered on top of a Layer 1 solution. Layer 2 protocols use the Layer 1 blockchain for network and security infrastructure.
+
+The data is to be ingested as an Ethereum [data source](./0045-DSRC-data_sourcing.md) but pointing at a different RPC endpoint.
+Hence this is in addition to and building upon [Ethereum data source](./0082-ETHD-ethereum-data-source.md).
+
+## Description
+
+In addition to listening to Ethereum events and reading from Ethereum contracts as described in [Ethereum data source](./0082-ETHD-ethereum-data-source.md) it will be possible for Vega nodes to listen to events from and read from other chains that implement Ethereum RPC and run EVM, in particular Ethereum L2.
```
Comment (davidsiska-vega) [inline protocol/0087-EVMD-eth-rpc-and-evm-data-source.md]: L2s - plural.

File: protocol/0087-EVMD-eth-rpc-and-evm-data-source.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+A new network parameter, a JSON list of network id / chain id / name one per Ethereum RPC and EVM chain will be used for setting supported L2. This is set via [governance](./0028-GOVE-governance.md).
+Name `blockchains.ethereumL2Configs`.
+Example value:
+
+```json
+{"configs": [{"network_id": 10, "chain_id": 10, "confirmations": 3, "name":"optimism"}]}
+```
+
+Duplicate values of `network_id`, `chain_id` or  `name` are not allowed (an update will be rejected at validation stage).
+Any update must always change the entire JSON (it's not possible to change individual entries).
```
Comment (davidsiska-vega) [inline protocol/0087-EVMD-eth-rpc-and-evm-data-source.md:30]: Only number of confirmations can be "changed", any other change is "removal" + "addition" in effect.

Comment (davidsiska-vega) [reply protocol/0087-EVMD-eth-rpc-and-evm-data-source.md:30]: In current minimal implementation, at proposal validation, check that only change is a) changing number of confirmation or b) adding another source. Add ACs to that effect.

**Review by wwestgarth — DISMISSED** [review]

**Review by Jiajia-Cui — APPROVED** [review]

**Pull request merged**